text/0000-global-packages.md

```diff
@@ -0,0 +1,195 @@
+
+- Delegation to project-local binaries where appropriate.
+- Pinning the version used to install a binary.
+
+This will greatly simplify the mental model needed to use Volta and reduce confusion around the package install process.
+
+# Details
+[details]: #details
+
+## Constraints
+
+1. **Global installs with package managers should work** We need to move away from our hard error interception of `npm i -g` and `yarn global add` and instead move to use those commands as the _primary_ way of installing global packages. This will allow users to continue with the familiar workflows they already use.
+2. **Global packages should continue to work when switching Node versions** While we currently handle this with siloed versions of Node, this is one of the key benefits that Volta provides over other Node managers and it should be preserved.
+3. **Global binaries should be able to call other global binaries** With a "standard" npm or Yarn setup, all of the tools that are globally installed are put onto the PATH directly, so they can all call each other without issue. We add the associated Node version to global tools, so we likely need to go through the shim logic, but that shouldn't stop tools from being able to call other tools (or themselves recursively).
+4. **Global libraries should be allowed** While not extremely useful, since they aren't generally accessible from arbitrary scripts, we should nevertheless allow users to install global packages that don't have any binaries. This is important for things like peer dependencies and `yeoman` generators.
+5. **Global packages should be able to `require` globally installed libraries** npm installs all global packages under a single `node_modules` directory, so when calling `require` from one package, it automatically can see the other packages that were installed globally. We should preserve this behavior as much as we can; the main difficulty I see here is native modules and differing associated Node versions.
+6. **Global packages should leverage Node module resolution** Our directory structure currently doesn't have any folders named `node_modules`, so the Node module resolution algorithm is completely unable to detect our packages, even from within a package's dependencies. There shouldn't be a need to completely reinvent the wheel here, we should do our best to leverage the existing resolution algorithm.
+
+## Overview
+
+The core of this approach is to incrementally satisfy the constraints to solve the outstanding issues. This will allow us to roll out improvements in steps, without having to make major, sweeping changes to the Volta implementation all at once. That flexibility will also allow us to iterate more quickly on the solution to make sure we cover the most common use-cases.
+
+## 0. Keep Existing Sandbox-per-Package Model
```

Comment:
Curious why this is zero-indexed?

Reply:
Mostly because it wasn’t a change needed, just a call-out of how we already meet the one constraint.

Reply:
PResentationally, maybe you could make that a little more obvious, like "Step 0 (done!) Keep existing Sandbox-per-Package Model or something like that. NBD tho!

Reply:
Added a "(Complete!)" to this line, good thinking!